src/sentry/api/invite_helper.py

```diff
@@ -177,8 +189,7 @@ def valid_request(self) -> bool:
     def accept_invite(self, user: User | None = None) -> OrganizationMember | None:
         om = self.om
 
-        if user is None:
-            user = self.request.user
+        user_id = self.invite.user_id if user is None else user.id
```

Comment:
Since we're messing with `AuthIdentity` (a control-only model) a few lines down anyway, we could just fetch the actual `User` object, especially if that makes the `create_audit_entry` call easier to handle. Still stuck with no `Request` object, though.